File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit=[]
        i,j=0,1
        for j in range(1,len(prices)):
            k=prices[j]-min(prices[i:j])
            if k>0:
                profit.append(k)
        if len(profit)!=0:
            return max(profit)
        return 0

File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& h) {
        int n=h.size();
        int i=0;
        int j=n-1;
        int maxarea=0;
        while(i<j)
        {
          int area=(j-i)*(min(h[i],h[j]));
          maxarea=max(maxarea,area);
         if(h[i]<h[j])
         {
            i++;
         }
         else
         {
            j--;
         }
        }
        return maxarea;
    }
};
